### ISSUE TITLE
`constructor` breaks syntax highlighting for rest of input

### ISSUE THREAD

**Issue by kdex** [atom/language-javascript#550]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Since GitHub, too, uses this grammar, check out the GitHub syntax highlighting here:

```js
while (true) {
	if (`${constructor}`) {
	}
} // Everything parsed below this line will be broken
function check(me, out) {
	what(happened);
}
```

For some reason, this highlights just fine:

```js
if (`${constructor}`) {
}
function check(me, out) {
	what(happened);
}
```

This, again, does not:

```js
if (true) {
	if (`${constructor}`) {
	}
} // Everything parsed below this line will be broken
function check(me, out) {
	what(happened);
}
```

### Steps to Reproduce

1. See examples above.

**Expected behavior:** Should highlight code after `constructor` just fine.

**Actual behavior:** It doesn't.

**Reproduces how often:** Always.

### Versions

I don't use atom, I've noticed this issue on GitHub.

### Additional Information

None

**Comment by kdex** [comment]
Is this the same issue as #425? My second example shows that `constructor` doesn't necessarily always break stuff.

**Comment by 50Wliu** [comment]
I'm closing this in favor as #425 as it's very likely that this will be fixed as well when #425 is fixed.

**Comment by kdex** [comment]
@50Wliu Thanks! I'll be following #425 then. :)
